This patch fixes HDL generation such that the Video and Keyboard components get a global clock connection.

diff --git a/src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java b/src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java
index 6fc6e11e2..a473f14a9 100644
--- a/src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java
+++ b/src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java
@@ -18,6 +18,8 @@ import com.cburch.logisim.fpga.gui.Reporter;
 import com.cburch.logisim.instance.Port;
 import com.cburch.logisim.instance.StdAttr;
 import com.cburch.logisim.prefs.AppPreferences;
+import com.cburch.logisim.std.io.Keyboard;
+import com.cburch.logisim.std.io.Video;
 import com.cburch.logisim.std.wiring.ClockHdlGeneratorFactory;
 import com.cburch.logisim.util.LineBuffer;
 import com.cburch.logisim.util.StringUtil;
@@ -395,7 +397,8 @@ public class AbstractHdlGeneratorFactory implements HdlGeneratorFactory {
                 String.format("Component \"%s\" in circuit \"%s\" has a gated clock connection!", compName, nets.getCircuitName()));
             gatedClock = true;
           }
-          if (hasClock && !gatedClock && Netlist.isFlipFlop(attrs)) {
+          if (hasClock && !gatedClock && (Netlist.isFlipFlop(attrs) || componentInfo.getComponent() instanceof Video
+              || componentInfo.getComponent() instanceof Keyboard)) {
             if (nets.requiresGlobalClockConnection()) {
               result.put(myPorts.getTickName(port), LineBuffer
                   .formatHdl("{{1}}{{<}}{{2}}{{>}}", clockNetName, ClockHdlGeneratorFactory.GLOBAL_CLOCK_INDEX));
